wtransport/src/stream.rs

```diff
@@ -81,17 +85,27 @@ impl SendStream {
     /// No new data can be written after calling this method. Locally buffered data is dropped, and
     /// previously transmitted data will no longer be retransmitted if lost. If an attempt has
     /// already been made to finish the stream, the peer may still receive all written data.
+    ///
+    /// # Panics
+    ///
+    /// If `reset` was already called.
     #[inline(always)]
-    pub fn reset(self, error_code: VarInt) {
+    pub fn reset(&mut self, error_code: VarInt) {
         self.0.reset(error_code);
     }
```

Comment:
While testing with my application, I realized how problematic taking ownership of a `SendStream` was (it's not compatible with `Pin`). Now, regardless of whether a `SendStream::{finish, stopped, reset}` all take mutable references. The downside is that calling `reset` twice, or `priority` after `reset` will panic.

Reply:
Yes, I agree with this changed, also required for `stopped` https://github.com/BiagioFesta/wtransport/issues/175